dataset/st_dataset.py

```diff
@@ -84,25 +86,75 @@ def __init__(self,
         self._test_set = test_set
         
     @property
-    def train_set(self) -> List[SummInstance]:
+    def train_set(self) -> Union[Generator[SummInstance, None, None], List]:
         if self._train_set is not None:
             return self._train_set
         else:
             print(f"{self.dataset_name} does not contain a train set, empty list returned")
             return list()
 
     @property
-    def dev_set(self) -> List[SummInstance]:
+    def dev_set(self) -> Union[Generator[SummInstance, None, None], List]:
         if self._dev_set is not None:
             return self._dev_set
         else:
             print(f"{self.dataset_name} does not contain a dev set, empty list returned")
             return list()
     
     @property
-    def test_set(self) -> List[SummInstance]:
+    def test_set(self) -> Union[Generator[SummInstance, None, None], List]:
         if self._test_set is not None:
             return self._test_set
         else:
             print(f"{self.dataset_name} does not contain a test set, empty list returned")
             return list()
+
+
+# Creating the train, dev and test splits from a dataset
+# the generated sets are 'train: ~.80', 'dev: ~.10', and 'test: ~10' in size
```

Comment:
Do we still do this if the dataset already has an original split?

Reply:
No we don't. The function is called only when the dataset has no splits. In our case, that would be Pubmedqa and Scisummnet.

Reply:
I don't see the need of making it a class method since only 2/10 datasets need to use it. Also, the alternative would most likely lead to refactoring how we set up our Dataset class.

Reply:
What do you think?

Reply:
> No we don't. The function is called only when the dataset has no splits. In our case, that would be Pubmedqa and Scisummnet.

I see, this sounds good to me.